models/spike_detector.cpp

```diff
@@ -139,7 +139,27 @@ void
 nest::spike_detector::set_status( const DictionaryDatum& d )
 {
   if ( d->known( names::precise_times ) )
-    user_set_precise_times_ = true;
+  {
+    has_user_set_precise_times_ = true;
+  }
+
+  // By setting precision value, precise_times is implicitly set.
+  if ( d->known( names::precision ) )
+  {
+
+    if ( d->known( names::precise_times ) )
+    {
+      if ( getValue< bool >( d, names::precise_times ) )
+      {
+        device_.set_precise( true, getValue< long >( d, names::precision ) );
+      }
+    }
```

Comment:
Yes, this was due to a confusion on my part. There is no reason to add this to the class.